collectd/files/nginx.conf

```diff
@@ -1,9 +1,20 @@
+{%- from "collectd/map.jinja" import collectd_settings with context %}
+{%- set nginx_settings = collectd_settings.plugins.nginx -%}
+
+#
+# DO NOT EDIT
+#
+# This file is managed by salt via {{ source }}
+# Modify the config that generates this file instead
+#
+
 LoadPlugin nginx
+
 <Plugin "nginx">
-  {{ 'URL "{}"'.format(url) if url else '' }}
-  {{ 'User "{}"'.format(user) if user else '' }}
-  {{ 'Password "{}"'.format(password) if password else '' }}
-  {{ 'VerifyPeer "{}"'.format(verifypeer) if verifypeer else '' }}
-  {{ 'VerifyHost "{}"'.format(verifyhost) if verifyhost else '' }}
-  {{ 'CACert "{}"'.format(cacert) if cacert else '' }}
+  URL "{{ nginx_settings.url }}"
+  User "{{ nginx_settings.user }}"
+  Password "{{ nginx_settings.password }}"
+  VerifyPeer "{{ nginx_settings.verifypeer }}"
+  VerifyHost "{{ nginx_settings.verifyhost }}"
+  CACert "{{ nginx_settings.cacert }}"
```

Comment:
If you are going to take out the ability to fall back on collectd defaults here, you should fix the defaults in the map.jinja file.

https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_nginx